src/test/java/org/jooq/lambda/SeqTest.java

```diff
@@ -988,7 +988,7 @@ public void testOnEmpty() throws X {
     }
 
     @SuppressWarnings("serial")
-    class X extends Exception {}
+    static class X extends RuntimeException {}
```

Comment:
Hmm, there was an implicit test to validate that things work for checked exceptions, too. I'll revert this part, rename `X` to `Checked` and add a new `Unchecked` exception for your test after mergin.